front/migrations/20241101_workspace_keys_incident.ts

```diff
@@ -0,0 +1,211 @@
+import { parse } from "csv-parse";
+import * as fs from "fs";
+import { exit } from "process";
+
+import { sendEmailWithTemplate } from "@app/lib/api/email";
+import { getMembers } from "@app/lib/api/workspace";
+import { Authenticator } from "@app/lib/auth";
+import { Workspace } from "@app/lib/models/workspace";
+import { getStripeClient } from "@app/lib/plans/stripe";
+import type { Logger } from "@app/logger/logger";
+import { makeScript } from "@app/scripts/helpers";
+
+const MIN_AMOUNT_FOR_REIMBURSEMENT = 1;
+
+const sendIncidentEmailToAdmins = async (
+  workspaceId: number,
+  cost_in_cents: number,
+  reimbursement_in_cents: number,
+  currency: string,
+  logger: Logger
+) => {
+  const message = {
+    from: {
+      name: "Dust team",
+      email: "team@dust.tt",
+    },
+    subject: `[Dust] Incident 2024/10/31`,
+    body: `<p>On October the 31st, due to a misconfiguration on our end, from 12:44 to 13:20 our API keys to OpenAI, Anthropic, Gemini and Mistral became unavailable to our systems triggering a fallback to your own API keys for assistants interactions within your workspace.</p>
+      <p>While it was great that we were able to serve your conversations on your own keys to avoid disruption, this is also not what you expect as a user. We estimated the tokens cost of the execution of these assistants to ${cost_in_cents / 100} ${currency.toUpperCase()}. We are applying a discount of ${reimbursement_in_cents / 100} ${currency.toUpperCase()} to your account to compensate for it.</p>
+      <p>Please reply to this email if you have any questions.</p>`,
+  };
+
+  const workspace = await Workspace.findByPk(workspaceId);
+  if (!workspace) {
+    // Should not happen but just in case
+    logger.error(`Workspace ${workspaceId} not found. Skipping email.`);
+    return;
+  }
+
+  const auth = await Authenticator.internalAdminForWorkspace(workspace.sId);
+
+  const { members: admins } = await getMembers(auth, { roles: ["admin"] });
```

Comment:
We were missing the `activeOnly` 🤦 so we probably sent the email to a bunch of revoked admins (cc @gabhubert)